deployment/ccip/changeset/cs_sync_usdc_domains_with_chains.go

```diff
@@ -0,0 +1,165 @@
+		}
+		if chainState.Timelock == nil {
+			return fmt.Errorf("missing timelock on %s", chain.String())
+		}
+		if chainState.ProposerMcm == nil {
+			return fmt.Errorf("missing proposerMcm on %s", chain.String())
+		}
+		if err = config.Validate(env.GetContext(), chain, chainState, c.MCMS != nil, c.ChainSelectorToUSDCDomain); err != nil {
+			return fmt.Errorf("USDC config for %s is not valid: %w", chain, err)
+		}
+	}
+	// Check that our input covers all chains that define USDC pools.
+	for chainSelector, chainState := range state.Chains {
+		if _, ok := c.USDCConfigsByChain[chainSelector]; !ok && chainState.USDCTokenPools != nil {
+			return fmt.Errorf("no USDC chain config defined for %s, which does support USDC", env.Chains[chainSelector])
+		}
+	}
+	return nil
+}
+
+// SyncUSDCDomainsWithChainsChangeset syncs domain support on specified USDC token pools with its chain support.
+// As such, it is expected that ConfigureTokenPoolContractsChangeset is executed before running this changeset.
+func SyncUSDCDomainsWithChainsChangeset(env deployment.Environment, c SyncUSDCDomainsWithChainsConfig) (deployment.ChangesetOutput, error) {
+	state, err := LoadOnchainState(env)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+	if err := c.Validate(env, state); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid SyncUSDCDomainsWithChainsConfig: %w", err)
+	}
+	readOpts := &bind.CallOpts{Context: env.GetContext()}
+
+	deployerGroup := NewDeployerGroup(env, state, c.MCMS)
+
+	for chainSelector, usdcChainConfig := range c.USDCConfigsByChain {
+		chain := env.Chains[chainSelector]
+		chainState := state.Chains[chainSelector]
+		writeOpts, err := deployerGroup.GetDeployer(chainSelector)
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to get transaction opts for %s", chain)
+		}
+
+		usdcTokenPool := chainState.USDCTokenPools[usdcChainConfig.Version]
+		supportedChains, err := usdcTokenPool.GetSupportedChains(readOpts)
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to fetch supported chains from USDC token pool with address %s on %s: %w", usdcTokenPool.Address(), chain, err)
+		}
+
+		domainUpdates := make([]usdc_token_pool.USDCTokenPoolDomainUpdate, 0)
+		for _, remoteChainSelector := range supportedChains {
+			remoteChainState := state.Chains[remoteChainSelector]
+			remoteUSDCChainConfig := c.USDCConfigsByChain[remoteChainSelector]
+			remoteUSDCTokenPool := remoteChainState.USDCTokenPools[remoteUSDCChainConfig.Version]
+
+			var desiredAllowedCaller [32]byte
+			remoteUSDCTokenPoolAddressBytes := remoteUSDCTokenPool.Address().Bytes()
+			for i, j := len(desiredAllowedCaller)-len(remoteUSDCTokenPoolAddressBytes), 0; i < len(desiredAllowedCaller); i, j = i+1, j+1 {
+				desiredAllowedCaller[i] = remoteUSDCTokenPoolAddressBytes[j]
+			}
```

Comment:
can you not use 
`  desiredAllowedCaller  := common.LeftPadBytes(remoteUSDCTokenPool.Address().Bytes(), 32),`

Reply:
https://github.com/smartcontractkit/chainlink/pull/16062/commits/8d1f3368c00ba86eb007e554b4aa949a31a9fb80